dali/test/python/test_dali_cpu_only.py

```diff
@@ -0,0 +1,872 @@
+    test_data_shape = [200]
+    def get_data():
+        out = [np.random.randint(0, 255, size = test_data_shape, dtype = np.uint8) for _ in range(batch_size)]
+        out[0][0] = 0
+        out[1][0] = 0
+        out[1][1] = 0
+        return out
+    data = fn.external_source(source = get_data)
+    processed, _ = fn.nonsilent_region(data)
+    pipe.set_outputs(processed)
+    pipe.build()
+    for _ in range(3):
+        pipe.run()
+
+def test_preemphasis_filter_cpu():
+    pipe = Pipeline(batch_size=batch_size, num_threads=4, device_id=-1)
+    test_data_shape = [200]
+    def get_data():
+        out = [np.random.ranf(size = test_data_shape).astype(dtype = np.float32) for _ in range(batch_size)]
+        return out
+    data = fn.external_source(source = get_data)
+    processed = fn.preemphasis_filter(data)
+    pipe.set_outputs(processed)
+    pipe.build()
+    for _ in range(3):
+        pipe.run()
+
+def test_power_spectrum_cpu():
+    pipe = Pipeline(batch_size=batch_size, num_threads=4, device_id=-1)
+    test_data_shape = [200]
+    def get_data():
+        out = [np.random.ranf(size = test_data_shape).astype(dtype = np.float32) for _ in range(batch_size)]
+        return out
+    data = fn.external_source(source = get_data)
+    processed = fn.power_spectrum(data)
+    pipe.set_outputs(processed)
+    pipe.build()
+    for _ in range(3):
+        pipe.run()
+
+def test_spectrogram_cpu():
+    pipe = Pipeline(batch_size=batch_size, num_threads=4, device_id=-1)
+    test_data_shape = [200]
+    def get_data():
+        out = [np.random.ranf(size = test_data_shape).astype(dtype = np.float32) for _ in range(batch_size)]
+        return out
+    data = fn.external_source(source = get_data)
+    processed = fn.spectrogram(data, nfft = 60, window_length = 50, window_step = 25)
+    pipe.set_outputs(processed)
+    pipe.build()
+    for _ in range(3):
+        pipe.run()
+
+def test_mel_filter_bank_cpu():
+    pipe = Pipeline(batch_size=batch_size, num_threads=4, device_id=-1)
+    test_data_shape = [200]
+    def get_data():
+        out = [np.random.ranf(size = test_data_shape).astype(dtype = np.float32) for _ in range(batch_size)]
+        return out
+    data = fn.external_source(source = get_data)
+    spectrum = fn.spectrogram(data, nfft = 60, window_length = 50, window_step = 25)
+    processed = fn.mel_filter_bank(spectrum)
+    pipe.set_outputs(processed)
+    pipe.build()
+    for _ in range(3):
+        pipe.run()
+
+def test_to_decibels_cpu():
+    pipe = Pipeline(batch_size=batch_size, num_threads=4, device_id=-1)
+    test_data_shape = [200]
+    def get_data():
+        out = [np.random.ranf(size = test_data_shape).astype(dtype = np.float32) for _ in range(batch_size)]
+        return out
+    data = fn.external_source(source = get_data)
+    spectrum = fn.spectrogram(data, nfft = 60, window_length = 50, window_step = 25)
+    mel = fn.mel_filter_bank(spectrum)
+    processed = fn.to_decibels(mel)
```

Comment:
you can run to_decibels on random data, no need to run the other ops that are already tested separately

Reply:
Done